queries/ecma/highlights.scm

```diff
@@ -0,0 +1,271 @@
+; Types
+
+; Javascript
+
+; Variables
+;-----------
+(identifier) @variable
+
+; Properties
+;-----------
+
+(property_identifier) @property
+(shorthand_property_identifier) @property
+
+; Special identifiers
+;--------------------
+
+((identifier) @constructor
+ (#match? @constructor "^[A-Z]"))
+
+((identifier) @constant
+ (#vim-match? @constant "^[A-Z_][A-Z\\d_]+$"))
+
+((shorthand_property_identifier) @constant
+ (#vim-match? @constant "^[A-Z_][A-Z\\d_]+$"))
+
+((identifier) @variable.builtin
+ (#vim-match? @variable.builtin "^(arguments|module|console|window|document)$"))
+
+((identifier) @function.builtin
+ (#eq? @function.builtin "require"))
+
+; Function and method definitions
+;--------------------------------
+
+(function
+  name: (identifier) @function)
+(function_declaration
+  name: (identifier) @function)
+(method_definition
+  name: (property_identifier) @method)
+
+(pair
+  key: (property_identifier) @method
+  value: (function))
+(pair
+  key: (property_identifier) @method
+  value: (arrow_function))
+
+(assignment_expression
+  left: (member_expression
+    property: (property_identifier) @method)
+  right: (arrow_function))
+(assignment_expression
+  left: (member_expression
+    property: (property_identifier) @method)
+  right: (function))
+
+(variable_declarator
+  name: (identifier) @function
+  value: (arrow_function))
+(variable_declarator
+  name: (identifier) @function
+  value: (function))
+
+(assignment_expression
+  left: (identifier) @function
+  right: (arrow_function))
+(assignment_expression
+  left: (identifier) @function
+  right: (function))
+
+; Function and method calls
+;--------------------------
+
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function: (member_expression
+    property: (property_identifier) @method))
+
+; Variables
+;----------
+
+(formal_parameters (identifier) @parameter)
+
+(formal_parameters
+  (rest_parameter
+    (identifier) @parameter))
+
+; ({ a }) => null
+(formal_parameters
+  (object_pattern
+    (shorthand_property_identifier) @parameter))
+
+; ({ a: b }) => null
+(formal_parameters
```

Comment:
Doesn't one of the nodes for parameters in javascript not exist in typescript?

Reply:
That's the exact problem. I guess @kyazdani42 said that he would prepare this PR for #1000 and 0.19.X parsers

Reply:
@kyazdani42 do you want to merge this one or do the change in/after #1000 ?

Reply:
hum they were included already anyway.
I'll do the changes in #1000, i'd prefer merging this first